metaflow/plugins/events_decorator.py

```diff
@@ -26,12 +26,21 @@ class TriggerDecorator(FlowDecorator):
     Additionally, you can specify the parameter mappings
     to map event payload to Metaflow parameters for the flow.
     ```
-    @trigger(event={'name':'foo', 'parameters':{'my_param': 'event_field'})
+    @trigger(event={'name':'foo', 'parameters':{'flow_param': 'event_field'}})
     ```
     or
     ```
-    @trigger(events=[{'name':'foo', 'parameters':{'my_param_1': 'event_field_1'},
-                     {'name':'bar', 'parameters':{'my_param_2': 'event_field_2'}])
+    @trigger(events=[{'name':'foo', 'parameters':{'flow_param_1': 'event_field_1'},
+                     {'name':'bar', 'parameters':{'flow_param_2': 'event_field_2'}])
+    ```
+
+    'parameters' can also be a list of strings and tuples like so:
+    ```
+    @trigger(event={'name':'foo', 'parameters':['common_name', ('flow_param', 'event_field')]})
```

Comment:
This is to support our current use case within Netflix. The issue has to do with implicit mapping which we do not want to allow within Netflix because of the potential of unintended consequences. The scenario is the following:
  - a user has a set of Parameters, a, b, and c and implicitly this maps to the signals a, b and c keys
  - the signal also contains d
  - the user adds d as a Parameter but *does not expect it to come from the signal*
With the current implementation, d would come from the signal. This use case could be for example that signals have certain unused values and a name collision happens.

What we want to do is make the mapping *explicit* and although we could do it with the existing syntax, it was long and cumbersome. After some internal discussion on slack (I'll have to find the thread), we settled on this syntax to have an explicit "shorthand" where the user would explicitly list the values that should come from the signal. The typical syntax would be: "parameters": ["value_1", "value_2", ...] and the tuple notation is to support renaming with that syntax as well.

And yes, this is not in the docs and would need to be added if this is something OSS wants to have as well.

Reply:
Makes perfect sense regarding the scenario. I'll try doing a quick test run with this today, but otherwise looks good to go :)